packages/desktop-client/src/components/reports/spreadsheets/master-spreadsheet.js

```diff
@@ -0,0 +1,432 @@
+        startNetWorth = balance;
+      }
+      endNetWorth = balance;
+
+      const change = last ? balance - amountToInteger(last.y) : 0;
+
+      const label = ChangeLabel(
+        income,
+        expense,
+        balance,
+        assets,
+        debt,
+        change,
+        totalExpenses,
+        totalIncome,
+        isCashFlow,
+        isNetWorth,
+        isIE,
+        isConcise,
+        x,
+      );
+
+      res.income.push({ x, y: integerToAmount(income), premadeLabel: label });
+      res.expenses.push({
+        x,
+        y: integerToAmount(expense),
+        premadeLabel: label,
+      });
+      res.change.push({
+        x,
+        y: integerToAmount(expense + income),
+        premadeLabel: label,
+      });
+      res.balances.push({
+        x,
+        y: integerToAmount(balance),
+        premadeLabel: label,
+      });
+      return res;
+    },
+    { expenses: [], income: [], change: [], balances: [] },
+  );
+
+  return {
+    graphData,
+    catData,
+    totalExpenses,
+    totalIncome,
+    hasNegative,
+    netWorth: endNetWorth,
+    totalChanges: endNetWorth - startNetWorth,
+    startNetWorth,
+  };
+}
+
+function ChangeLabel(
+  income,
+  expense,
+  balance,
+  assets,
+  debt,
+  change,
+  totalExpenses,
+  totalIncome,
+  isCashFlow,
+  isNetWorth,
+  isIE,
+  isConcise,
+  x,
+) {
+  if (isNetWorth) {
+    return (
+      <div>
+        <div style={{ marginBottom: 10 }}>
+          <strong>{d.format(x, 'MMMM yyyy')}</strong>
+        </div>
+        <div style={{ lineHeight: 1.5 }}>
+          <AlignedText left="Assets:" right={integerToCurrency(assets)} />
+          <AlignedText left="Debt:" right={integerToCurrency(debt)} />
+          <AlignedText
+            left="Net worth:"
+            right={<strong>{integerToCurrency(balance)}</strong>}
+          />
+          <AlignedText left="Change:" right={integerToCurrency(change)} />
+        </div>
+      </div>
+    );
+  } else if (isCashFlow) {
+    return (
+      <div>
+        <div style={{ marginBottom: 10 }}>
+          <strong>
+            {d.format(x, isConcise ? 'MMMM yyyy' : 'MMMM d, yyyy')}
+          </strong>
+        </div>
+        <div style={{ lineHeight: 1.5 }}>
+          <AlignedText left="Income:" right={integerToCurrency(income)} />
+          <AlignedText left="Expenses:" right={integerToCurrency(expense)} />
+          <AlignedText
+            left="Change:"
+            right={<strong>{integerToCurrency(income + expense)}</strong>}
+          />
+          <AlignedText left="Balance:" right={integerToCurrency(balance)} />
+        </div>
+      </div>
+    );
+  } else if (isIE) {
+    return (
+      <div>
+        <div style={{ marginBottom: 10 }}>
+          <strong>{d.format(x, 'MMMM yyyy')}</strong>
+        </div>
+        <div style={{ lineHeight: 1.5 }}>
+          {income ? (
+            <AlignedText left="Income:" right={integerToCurrency(income)} />
+          ) : (
+            ''
+          )}
```

Comment:
This formats a bit better with Prettier, fyi

```suggestion
          ) : null}
```